src/main/java/org/nuxeo/audit/storage/stream/StreamAuditStorageWriter.java

```diff
@@ -0,0 +1,148 @@
+                       .build();
+    }
+
+    public class AuditStorageLogWriterComputation extends AbstractComputation {
+        protected final int batchSize;
+
+        protected final int batchThresholdMs;
+
+        protected final List<String> jsonEntries;
+
+        public AuditStorageLogWriterComputation(String name, int batchSize, int batchThresholdMs) {
+            super(name, 1, 0);
+            this.batchSize = batchSize;
+            this.batchThresholdMs = batchThresholdMs;
+            jsonEntries = new ArrayList<>(batchSize);
+        }
+
+        @Override
+        public void init(ComputationContext context) {
+            log.debug(String.format("Starting computation: %s reading on: %s, batch size: %d, threshold: %dms",
+                    COMPUTATION_NAME, STREAM_NAME, batchSize, batchThresholdMs));
+            context.setTimer("batch", System.currentTimeMillis() + batchThresholdMs);
+        }
+
+        @Override
+        public void processTimer(ComputationContext context, String key, long timestamp) {
+            writeJsonEntriesToAudit(context);
+            context.setTimer("batch", System.currentTimeMillis() + batchThresholdMs);
+        }
+
+        @Override
+        public void processRecord(ComputationContext context, String inputStreamName, Record record) {
+            try {
+                jsonEntries.add(getJsonEntryFromBytes(record.data));
+            } catch (NuxeoException e) {
+                log.error("Discard invalid record: " + record, e);
+                return;
+            }
+            if (jsonEntries.size() >= batchSize) {
+                writeJsonEntriesToAudit(context);
+            }
+        }
+
+        @Override
+        public void destroy() {
+            log.debug(String.format("Destroy computation: %s, pending entries: %d", COMPUTATION_NAME,
+                    jsonEntries.size()));
+        }
+
+        /**
+         * Store JSON entries in the Directory Audit Storage
+         */
+        protected void writeJsonEntriesToAudit(ComputationContext context) {
+            if (jsonEntries.isEmpty()) {
+                return;
+            }
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("Writing %d log entries to the directory audit storage %s.", jsonEntries.size(),
+                        DirectoryAuditStorage.NAME));
+            }
+            NXAuditEventsService audit = (NXAuditEventsService) Framework.getRuntime()
+                                                                         .getComponent(NXAuditEventsService.NAME);
+            DirectoryAuditStorage storage = (DirectoryAuditStorage) audit.getAuditStorage(DirectoryAuditStorage.NAME);
+            storage.append(jsonEntries);
+            jsonEntries.clear();
+            context.askForCheckpoint();
+        }
+
+        protected String getJsonEntryFromBytes(byte[] data) {
+            try {
+                return new String(data, "UTF-8");
```

Comment:
Inline this in the caller method, to avoid the double exception catch/rethrow.
Use `java.nio.charset.StandardCharsets.UTF_8` (as a static import) instead of `"UTF-8"`